services/director-v2/tests/unit/test_modules_dynamic_sidecar_docker_api.py

```diff
@@ -63,27 +60,31 @@ def network_config(simcore_services_network_name: str) -> Dict[str, Any]:
 
 @pytest.fixture
 async def ensure_swarm_network(
-    loop: BaseEventLoop, network_config: Dict[str, Any]
+    loop: BaseEventLoop,
+    network_config: Dict[str, Any],
+    docker_client: aiodocker.docker.Docker,
+    async_docker_swarm: None,
 ) -> None:
     network_id = None
-    async with docker_client() as client:
-        try:
-            network_id = await create_network(network_config)
-            yield
-        finally:
-            if network_id is not None:
-                docker_network = await client.networks.get(network_id)
-                assert await docker_network.delete() is True
+    try:
+        network_id = await docker_api.create_network(network_config)
+        yield
+    finally:
+        if network_id is not None:
+            docker_network = await docker_client.networks.get(network_id)
+            assert await docker_network.delete() is True
 
 
 @pytest.fixture
 async def cleanup_swarm_network(
-    loop: BaseEventLoop, simcore_services_network_name: str
+    loop: BaseEventLoop,
+    simcore_services_network_name: str,
+    docker_client: aiodocker.docker.Docker,
+    async_docker_swarm: None,
 ) -> None:
     yield
-    async with docker_client() as client:
-        docker_network = await client.networks.get(simcore_services_network_name)
-        assert await docker_network.delete() is True
+    docker_network = await docker_client.networks.get(simcore_services_network_name)
+    assert await docker_network.delete() is True
```

Comment:
same question and same remark.

Reply:
The network is only created for this test. I see no reason why not to remove it.
There is a fixture, only to guarantee that if the test fails, the test will not be pending.

Reply:
all good concerning the network, I did not see that one.

but the other one, why not cleaning the networks/volumes created during swarm mode anyway at the end of the swarm? that would nicely clean up.
Especially since I think your problem is probably linked with the networks... these are the one that usually are taking time to remove.